### PR TITLE
[Hub Generated] Review request for Search to add version preview/1.0

### PR DESCRIPTION
**Pull request by ambientlight**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/ambientlight/azure-rest-api-specs/dev-maps-Search-fix-geojson...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by daniel-rocha** [comment]
@ambientlight  this is failing model validation, can you please look into that?

**Comment by ambientlight** [comment]
> @ambientlight this is failing model validation, can you please look into that?

I'm not yet sure how to fix this. This seems like new validation check (the validation command I use locally didn't show this), but either I am not understanding this or I am not agreeing with this check. 

Model validation is failing at https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1225399&view=logs&j=14c5c170-9612-5b44-a650-13d3ca524cf8&t=0b3a3337-12e3-5f9b-9f15-b3914db18063&l=30

with: `Data is valid against more than one schema from 'oneOf'`, while checking the sample, which most likely can be explained by the fact that [GeoJsonCircleFeature](https://review.docs.microsoft.com/en-us/rest/api/documentation-preview/search/post-search-inside-geometry?view=azure-rest-preview&branch=result_18d0ce51-53c6-42c8-b773-880af46a6f4b#geojsoncirclefeature) and [GeoJsonPolygonFeature](https://review.docs.microsoft.com/en-us/rest/api/documentation-preview/search/post-search-inside-geometry?view=azure-rest-preview&branch=result_18d0ce51-53c6-42c8-b773-880af46a6f4b#geojsonpolygonfeature) have the same discriminator type (type: "Feature"), which having different internal shape (circle extension inside feature or polygon inside feature).

At this point I'm not yet sure how to resolve it.

@alankashiwa, @alextts627, @daniel-rocha: Please confirm whether geojson serialization/deserialization is valid as result of this PR updates for your languages. I was able to get a valid codegen in C# that will work for all geojson usecases:

1. ListPolygons returning Polygon geometry.
2. ListPolygons returning MultiPolygon geometry.
3. SearchAlongRoute was working before this PR. (untouched and it wasn't polymorphic)
4. SearchInsideGeometry with polygon geometry as param.
5. SearchInsideGeometry with geometry collection containing polygon geometry.
6. SearchInsideGeometry with feature collection with both Polygon feature and Point(Circle) feature. **Here I needed to add additional fix in convenience layer** since codegen will not have properties property in GeoJson feature as required, and LBS will fail on validation if it is not specified.

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 1]
```diff
@@ -1839,7 +1839,63 @@
...
+          "description": "Properties can contain any additional metadata about the `Feature`. Value can be any JSON object or a JSON null value",
+          "type": "object"
+        }
+      },
+      "required": [
+        "geometry",
+        "properties"
+      ]
+    },
+    "GeoJsonMultiPolygonFeature": {
```
Comment (alankashiwa) [inline specification/maps/data-plane/Search/preview/1.0/search.json:1870]: The definition should be the same as `GeoJsonPolygonFeature` except the `description` and `properties:geometry` parts.

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 2]
```diff
@@ -2714,6 +2757,158 @@
...
+                "description": "`GeoJSON FeatureCollection` object.",
+                "name": "GeoJsonFeatureCollection"
+              }
+            ]
+          }
+        }
+      }
+    },
+    "GeoJsonCircleFeature": {
+      "description": "A valid `GeoJSON Point` geometry type. Please refer to [RFC 7946](https://tools.ietf.org/html/rfc7946#section-3.1.2) for details.",
```
Comment (alankashiwa) [inline specification/maps/data-plane/Search/preview/1.0/search.json:2802]: Should describe the `Circle` feature

File: specification/maps/data-plane/Search/preview/1.0/search.json [hunk 3]
```diff
@@ -1839,7 +1839,63 @@
...
+      "allOf": [
+        {
+          "$ref": "../../../Common/preview/1.0/common.json#/definitions/GeoJsonFeature"
+        }
+      ],
+      "properties": {
+        "geometry": {
+          "$ref": "../../../Common/preview/1.0/common.json#/definitions/GeoJsonPolygon"
+        },
+        "properties": {
```
Comment (alankashiwa) [inline specification/maps/data-plane/Search/preview/1.0/search.json:1859]: `properties` is already declared the parent schema.

**Pull request closed without merging**